codegen/templates/endpoint.tmpl

```diff
@@ -50,21 +52,20 @@ func Handle{{title .Name}}Request(
 	}
 	{{end}}
 
-	headers := map[string]string{}
```

Comment:
A workflow is transport agnostic. we dont want to regress back to having http specific workflows

It's interface should not include `http.Header`, this means it should have either `zanzibar.Header` or `map[string]string`